### DISCUSSION TITLE
import =

### DISCUSSION THREAD

**Discussion by SalathielGenese** [babel/babel#12435] [Ideas]
Following #7062, here is my use case.

I have some functions, with some specific parameter type, which types I scope under a namespace of the same name as the function. Therefore, I am not sure how this error insight helps me:

> `import =` is not supported by @babel/plugin-transform-typescript
api_1  | Please consider using `import <moduleName> from '<moduleName>';` alongside Typescript's --allowSyntheticDefaultImports option.

PS: I do not think my use case is a design pattern issue. My file contains a number of utility functions, to small to lie in a single place, with various respective type not to pollute the outer namespace.

```typescript
import A = doSomething.A;
import B = doSomething.B;

function doSomething( a: A, b: B ) {
  // TODO: Provide implementation
}
export module doSomething {
  export type A = string | number | { a: string | number };
  export type B = string | { b: string, bb: () => void } | B[];
}
```

**Comment by JLHwung** [comment]
See https://github.com/babel/babel/issues/12345#issuecomment-725585579
